.github/workflows/cd.yaml

```diff
@@ -38,14 +38,98 @@ jobs:
+        with:
+          go-version: '1.15'
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set current date as env variable
+        run: echo "TODAY=$(date +'%y-%m-%d')" >> $GITHUB_ENV
+      - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-${{ github.run_number }}
+          SOURCE_LOG_1: s3://brim-sampledata/wrccdc/zeek-logs/files.log.gz
+          SPACE: files
+          ZQD_DATA_URI: s3://zq-ci-test/${{ env.TODAY }}.${{ github.run_number }}
+        run: |
+          set -x
+          make install
+          aws eks --region us-east-2 update-kubeconfig --name $CLUSTER
+          kubectl config set-context --current --namespace=$NAMESPACE
+          kubectl scale --replicas=4 deployment/worker-zqd
+          kubectl port-forward svc/recruiter-zqd 8020:9867 &
+          kubectl port-forward svc/root-zqd 9867:9867 &
+          sleep 2 # wait for port-forwards to complete
+          time zapi ls
+          time zapi new -k archivestore -d $ZQD_DATA_URI $SPACE
+          # Post an archive of about 1.6 GB to run test queries.
+          time zapi -s $SPACE post $SOURCE_LOG_1
+          # There should be 4 workers running now.
+          [[ 4 == $(curl http://localhost:8020/recruiter/listfree | jq -r ".workers" | jq length) ]]     
+          time zapi -s $SPACE get -workers 1 -t '"FgzbkQ15TOoOENn1lb" | sort tx_hosts'
+          time zapi -s $SPACE get -workers 2 -t '"FgzbkQ15TOoOENn1lb" | sort tx_hosts'
+          time zapi -s $SPACE get -workers 4 -t '"FgzbkQ15TOoOENn1lb" | sort tx_hosts'
+          time zapi -s $SPACE get -workers 4 -t 'count()' > count.tzng
+          time zapi -s $SPACE get -workers 4 -t \
+            '* | count() by tx_hosts | sort -r count | head 5' > countbytxhost.tzng
+          diff count.tzng k8s/testdiff/count.tzng
+          diff countbytxhost.tzng k8s/testdiff/countbytxhost.tzng
+      - name: Cleanup on success or failure
+        if: ${{ always() }}
+        env:
+          NAMESPACE: ci-${{ github.run_number }}
+        run: kubectl delete ns $NAMESPACE
+  eks-master-deploy:
+    # Deploys this build to ci-master namespace so it will be available
+    # for manual testing. Only deploy if previous jobs succeed.
+    needs: eks-test-zapi-1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-master
```

Comment:
I know care was taken with the separate namespaces so concurrent runs shouldn't step on each other. However, at this point it seems to converge on a single namespace. Have we tested and/or know what will happen if two concurrent runs reach this step at the same time? I'm hopeful the underlying K8s operations are all atomic enough that one of them will consistently "win" and cancel the other out, as opposed to some kind of race causing deadlock or inconsistency, but figured I'd ask.

Reply:
That is a good question: I think Helm will be alright with a second install starting during another install that is in-progress. We still should keep an eye on it, in case it can fail in some way I have not seen in my testing so far. Since it is our test environment, it is a good place to learn if and when collisions of this type can happen.